components/compliance-service/ingest/ingestic/ingestic.go

```diff
@@ -43,19 +43,34 @@ func (backend *ESClient) addDataToIndexWithID(ctx context.Context,
 }
 
 // This method will support adding a document with a specified id
-func (backend *ESClient) upsertComplianceRunInfo(ctx context.Context, mapping mappings.Mapping, id string, runDateTime time.Time) error {
+func (backend *ESClient) upsertComplianceRunInfo(ctx context.Context, mapping mappings.Mapping, runInfo relaxting.ESComplianceRunInfo, runDateTime time.Time) error {
 	runDateTimeAsString := runDateTime.Format(time.RFC3339)
 
 	script := elastic.NewScript("ctx._source.last_run = params.rundate").Param("rundate", runDateTimeAsString)
 
 	_, err := backend.client.Update().
 		Index(mapping.Index).
-		Id(id).
+		Id(runInfo.NodeID).
 		Script(script).
 		Upsert(map[string]interface{}{
-			"node_uuid": id,
-			"first_run": runDateTime,
-			"last_run":  runDateTime,
+			"node_uuid":             runInfo.NodeID,
+			"resource_uuid":         runInfo.ResourceId,
+			"resource_type":         runInfo.ResourceType,
+			"status":                runInfo.Status,
+			"first_run":             runInfo.FirstRun,
+			"last_run":              runInfo.LastRun,
+			"platform_with_version": runInfo.PlatFormWithVersion,
+			"platform":              runInfo.Platform,
+			"control_tag":           runInfo.ControlTags,
+			"chef_server":           runInfo.ChefServer,
+			"organization":          runInfo.Organization,
+			"controls":              runInfo.Controls,
```

Comment:
What happenes if the new control is run on that node, at the time of updating last run info.